File: src/category_strings/q1_make_parenthesis_valid/MakeParenthesisValid.py
from utils.LogUtil import LogUtil;

class MakeParenthesisValid:

    # Write your code here
    #
    # Remove the minimum number of parentheses so that the resulting parenthesis string is valid 
    # (See README for details) 
    # @param s - input string containing ASCII characters
    # @return - ***any*** valid string. There may be more than 1 solution.
    def make_parenthesis_valid(self, s: str) -> str:
            return s;

    def test(self):
        # Change these values to manually test your program
        str_input = "y(e(e(h))a)w)";
        accepted_responses = set(["y(e(e(h))a)w", "y(e(e(h)a)w)", "y(e(e(h))aw)"])
        actual_response = self.make_parenthesis_valid(str_input);

        LogUtil().log_results_set("1: Make Parenthesis Valid", accepted_responses, actual_response);

File: src/category_strings/q1_make_parenthesis_valid/MakeParenthesisValid.js
const LogUtil = require('../../utils/LogUtil.js');

/**
 * Remove the minimum number of parentheses so that the resulting parenthesis string is valid 
 * See README for Details
 * @param {string} s - input string containing ASCII characters
 * @return {string} ***any*** valid string. There may be more than 1 solution.
 */
var makeParenthesisValid = function(s) {
    // Write your code here
    return s;
}

var test = function() {
    // Change these values to manually test your program
    let strInput = "y(e(e(h))a)w)";
    let acceptedResponses = ["y(e(e(h))a)w", "y(e(e(h)a)w)", "y(e(e(h))aw)"]
    let actualResponse = makeParenthesisValid(strInput);
    new LogUtil().logResultsSet("1: Make Parenthesis Valid", acceptedResponses, actualResponse);
}

module.exports = test;